config/src/test/java/io/confluent/common/config/ConfigDefTest.java

```diff
@@ -275,4 +285,79 @@ private void testValidators(Type type, ConfigDef.Validator validator, Object def
       }
     }
   }
+
+  @Test
+  public void testClassLoading() throws IOException {
+    ClassLoader systemLoader = ClassLoader.getSystemClassLoader();
+
+    // test with system classloader
+    String configName = "type.class.property";
+    String configNameDefault = Password.class.getName();
+
+    ConfigDef configDef = new ConfigDef();
+    configDef.define(configName, Type.CLASS, configNameDefault,
+        ConfigDef.Importance.MEDIUM, "system class");
+
+    AbstractConfig config = new AbstractConfig(configDef, new HashMap<>());
+
+    Class<?> systemKlass = config.getClass(configName);
+    assertEquals(systemLoader, systemKlass.getClassLoader());
+
+    // test with custom, child-first, classloader
+    Set<String> allowedKlasses = Collections.singleton(configNameDefault);
+    List<URL> pluginUrls = Collections.singletonList(Paths.get("target/classes/").toUri().toURL());
+    URLClassLoader pluginLoader = new PluginClassLoader(pluginUrls.toArray(new URL[0]), systemLoader, allowedKlasses);
+
+    Thread.currentThread().setContextClassLoader(pluginLoader);
```

Comment:
I think you should swap back out of this loader at the end of the test (or in a finally block).

Reply:
Good idea.